### ISSUE TITLE
Pause/Cancel training and continue later on …

### ISSUE THREAD

**Issue by fooness** [tensorflow/tensor2tensor#1266]
Dear community, I’m new to T2T … thank you for providing the framework and the documentation! I was following a tutorial to start training `transformer` with `languagemodel_ptb10k` for `10000` `train_steps` (with `100` `eval_steps`).

Is it possible to pause/cancel the training at some point and continue training at some other point (like tomorrow)?

Here’s the commands I used:

`!pip install tensor2tensor`

```
from tensor2tensor import problems

PROBLEM = 'languagemodel_ptb10k'
TMP_DIR = './temp'
DATA_DIR = './data'

t2t_problem = problems.problem(PROBLEM)
t2t_problem.generate_data(DATA_DIR, TMP_DIR)
```

```
from tensor2tensor.utils import registry
from tensor2tensor import models

PROBLEM = 'languagemodel_ptb10k'
MODEL = 'transformer'
HPARAMS = 'transformer_base'

TRAIN_DIR = './model'
DATA_DIR = './data'
```

```
from tensor2tensor.utils.trainer_lib import create_hparams

hparams = create_hparams(HPARAMS)

hparams.batch_size = 1024
hparams.learning_rate_warmup_steps = 45000
hparams.learning_rate = .4
```

```
from tensor2tensor.utils.trainer_lib import create_run_config, create_experiment

RUN_CONFIG = create_run_config(
    model_dir=TRAIN_DIR,
    model_name=MODEL
)

tensorflow_exp_fn = create_experiment(
        run_config=RUN_CONFIG,
        hparams=hparams,
        model_name=MODEL,
        problem_name=PROBLEM,
        data_dir=DATA_DIR, 
        train_steps=10000,
        eval_steps=100
    )

tensorflow_exp_fn.train_and_evaluate()
```

Thank you very much!

**Comment by mehmedes** [comment]
You can stop training hitting ```ctrl+c``` anytime. T2T will always resume training from your last checkpoint. Simply increase the number of  ```train_steps``` and T2T will continue to train your model untill reaching the desired number of steps.

**Comment by mehmedes** [comment]
Yes, there's a user group:
https://groups.google.com/forum/m/#!forum/tensor2tensor
Have you tried training your language model without special preprocessing? Does your loss decrease? Just feed the network with your training data and see how things develop. You may get reasonable results right away.
You can always stop training and check your model by inferring in interactive mode:
```
t2t-decoder \
  --data_dir=$DATA_DIR \
  --problem=$PROBLEM \
  --model=$MODEL \
  --hparams_set=$HPARAMS \
  --output_dir=$TRAIN_DIR \
  --decode_hparams="beam_size=$BEAM_SIZE,alpha=$ALPHA" \
--decode_interactive
```

**Comment by fooness** [comment]
@mehmedes thank you again, I just posted to the user group forum.

We haven’t tried training the language model without special preprocessing. Think we’re still a little confused on where the training data comes in place.

I think we understand that `transformer` is the training process/method that is being used—which was described in some paper on arxiv maybe, same goes for `attention_lm` I think.

The problem `languagemodel_ptb10k` is the ptb10k dataset, I guess, kind of prepared for the language modeling task. So maybe I’d need to replace this problem with our training data, which then probably needs to be prepared somewhere.

Sorry for the confusion … we’re having lots of it, but it’s getting better and better.
